pil-analyzer/src/type_inference.rs

```diff
@@ -145,10 +206,10 @@ impl TypeChecker {
                 continue;
             };
 
-            let (_, declared_type) = self.declared_types[&name].clone();
+            let declared_type = self.declared_types[&name].clone();
```

Comment:
Please abstract this way of access. It could just return a TypeScheme (and panic for struct because we don't get here for structs).